### PR TITLE
feat(backend): [do-not-review] Allow networkless OAT verification

### PR DESCRIPTION
**Pull request by wobsoriano**
## Checklist

- [ ] `pnpm test` runs as expected.
- [ ] `pnpm build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Summary by CodeRabbit

* **New Features**
  * Added support for OAuth JWT access tokens (at+jwt / application/at+jwt) across authentication flows.
  * OAuth tokens now undergo JWT-format and signature verification and are parsed for scopes and expiration.

* **Tests**
  * Expanded test coverage for OAuth JWT detection, parsing, expiry behavior, and verification edge cases.

* **Chores**
  * Added a changeset documenting a minor backend package bump.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: packages/backend/src/jwt/assertions.ts [hunk 1]
```diff
@@ -61,6 +61,21 @@ export const assertHeaderType = (typ?: unknown) => {
...
+};
+
+export const assertOAuthHeaderType = (typ?: unknown) => {
+  if (!isOAuthAccessTokenJwt(typ)) {
+    throw new TokenVerificationError({
+      reason: TokenVerificationErrorReason.TokenInvalid,
+      message: `Invalid OAuth JWT type ${JSON.stringify(typ)}. Expected "at+jwt" or "application/at+jwt".`,
+    });
+  }
+};
```
[the remark this answers is not in the document]
Comment (wobsoriano) [reply packages/backend/src/jwt/assertions.ts:81]: added doc for `OAUTH_ACCESS_TOKEN_TYPES`, kept the others. They're all internal and the function name is descriptive.

File: packages/backend/src/api/resources/IdPOAuthAccessToken.ts [hunk 2]
```diff
@@ -30,4 +32,23 @@ export class IdPOAuthAccessToken {
...
+      payload.sub,
+      oauthPayload.scp ?? oauthPayload.scope?.split(' ') ?? [],
+      false,
+      null,
+      payload.exp * 1000 <= Date.now() - clockSkewInMs,
+      payload.exp,
+      payload.iat ?? 0,
+      payload.iat ?? 0,
+    );
+  }
```
[the remark this answers is not in the document]
Comment (wobsoriano) [reply packages/backend/src/api/resources/IdPOAuthAccessToken.ts:64]: Updated to minimal doc
